cloud-minecraft/cloud-brigadier/src/main/java/cloud/commandframework/brigadier/NativeCompletion.java

```diff
@@ -0,0 +1,116 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+package cloud.commandframework.brigadier;
+
+import cloud.commandframework.Completion;
+import com.mojang.brigadier.LiteralMessage;
+import com.mojang.brigadier.Message;
+import org.apiguardian.api.API;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * A suggestion which gives native minecraft chat component as description, used for brigadier support
+ * @since 1.9.0
+ */
+public interface NativeCompletion extends Completion {
+
+    /**
+     * Gives a native chat component tooltip for suggestion
+     * @return chat component
+     */
+    @NonNull Message tooltip();
+
+    /**
+     * Returns a new native suggestion with a new tooltip
+     * @param tooltip the new tooltip message
+     * @return a new suggestion instance with this tooltip
+     */
+    @NonNull NativeCompletion withTooltip(@NonNull Message tooltip);
+
+    /**
+     * Gives a suggestion instance with native brigadier message as description
+     * @param suggestion the suggestion
+     * @param description the plain text as description
+     * @return a new suggestion instance
+     */
+    static @NonNull NativeCompletion of(@NonNull final String suggestion, @NonNull final String description) {
+        return new SimpleCompletion(suggestion, new LiteralMessage(description));
+    }
+    /**
+     * Gives a suggestion instance with native brigadier message as description
+     * @param suggestion the suggestion
+     * @param description the rich text as description
+     * @return a new suggestion instance
+     */
+    static @NonNull NativeCompletion of(@NonNull final String suggestion, @NonNull final Message description) {
+        return new SimpleCompletion(suggestion, description);
+    }
+    /**
+     * Gives a suggestion instance with native brigadier message as description
+     * if the description object is a brigadier message
+     * @param suggestion the suggestion
+     * @param description the rich text as description
+     * @return a new suggestion instance
+     */
+    static @NonNull Completion ofUnknown(@NonNull final String suggestion, @Nullable final Object description) {
```

Comment:
no way, this is nasty

Reply:
i thought this to be a way if they don't have brigadier dependency, but better I'll think of a way to add support for adventure for such reasons